conda/recipes/ptxcompiler/meta.yaml

```diff
@@ -13,16 +13,22 @@ source:
 
 build:
   number: {{ GIT_DESCRIBE_NUMBER }}
-  string: cuda_{{ cuda_version }}_py{{ py_version }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
   script: {{ PYTHON }} -m pip install . -vv
 
 requirements:
   build:
+    - {{ compiler("c") }} 9
+    - {{ compiler("cxx") }} 9
```

Comment:
Curious why this is pinning to GCC 9? Also if we do want this, we may want to handle this a different way.

Reply:
As we are using GCC 9 to build the RAPIDS libs, I was thinking that using the same GCC version could be a good idea. WDYT?

Reply:
PR updated with this. What's the difference between both?

Reply:
This is just [the recommended way for constraining compiler versions]( https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#compiler-versions ).

Also this is used in `conda-build` and [conda-forge globally]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/f69f0c07872a96bae32536fadeb6fcf195c86d1e/recipe/conda_build_config.yaml#L11 ). So specifying compiler versions in `conda_build_config.yaml` naturally overrides the logic in those places.

One other difference is [`compiler("c")` may not mean GCC in some contexts]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/f69f0c07872a96bae32536fadeb6fcf195c86d1e/recipe/conda_build_config.yaml#L7 ). For example MSVC is used on Windows and Clang is used on macOS. If we started building on other platforms, we would want to only apply this constraint on Linux where GCC is used and possibly use other constraints elsewhere. Admittedly one could still add selectors in `meta.yaml` to affect this change, but it is cleaner to have all this logic in `conda_build_config.yaml` where the compilers themselves are specified.